src/draw/transformations.typ

```diff
@@ -3,24 +3,57 @@
 #import "/src/vector.typ"
 #import "/src/util.typ"
 
-#let rotate(angle) = {
-  (
-    ctx => {
-      ctx.transform = matrix.mul-mat(ctx.transform, if type(angle) == "angle" {
-        matrix.transform-rotate-z(-angle)
-      } else if type(angle) == "dictionary" {
-        matrix.transform-rotate-xyz(
-          -angle.at("x", default: 0deg),
-          -angle.at("y", default: 0deg),
-          -angle.at("z", default: 0deg),
-        )
-      } else {
-        panic("Invalid angle format '" + repr(angle) + "'")
-      })
-      
-      return (ctx: ctx)
-    },
-  )
+/// Sets the transformation matrix
+///
+/// - mat (none,matrix): The 4x4 transformation matrix to set. If `none` is
+///   passed, the transformation matrix is set to the identity matrix (
+///   `matrix.ident()`).
+#let transform(mat) = {
```

Comment:
I would prefer the function name `set-transformation` given it sets the current transformation matrix. `transform` suggests that it applies the given matrix to the current transformation